### PR TITLE
power_nvram: make sure qemu-kvm works well while raw strings are passed

### PR DESCRIPTION
**Pull request by MiriamDeng**
ID:1935545

Signed-off-by: MiriamDeng <mdeng@redhat.com>

### CODE REVIEW

**Comment by MiriamDeng** [comment]
Test result, 
(1/3) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.power_nvram.big_size: PASS (6.82 s)
 (2/3) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.power_nvram.bigger_size: PASS (127.36 s)
 (3/3) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.power_nvram.normal_size: PASS (77.88 s)

**Comment by MiriamDeng** [comment]
@PaulYuuu @maxujun could you help to review this patch when you are free ? Thanks a lot.

File: qemu/tests/cfg/power_nvram.cfg [hunk 1]
```diff
@@ -0,0 +1,22 @@
+- power_nvram:
+    type = power_nvram
+    start_vm = yes
```
Comment (maxujun) [inline qemu/tests/cfg/power_nvram.cfg:3]: Can set this "start_vm = no" and delete "start_vm = no" for big_size and bigger_size

Comment (MiriamDeng) [reply qemu/tests/cfg/power_nvram.cfg:3]: Thanks for your reviewing but we need to set it as "yes" since the last case (normal size) need it.

Comment (PaulYuuu) [reply qemu/tests/cfg/power_nvram.cfg:3]: I agree with @maxujun , 2/3 use `start_vm = no`, only one use ` = yes`. So we can define `start_vm = no` as the public parameter, just add `start_vm = yes` in the specific variant.

File: qemu/tests/power_nvram.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        try:
+            vm.create(params=params)
+        except virt_vm.VMCreateError as e:
+            output = e.output
+            error_context.context("Check the expected error message: %s"
+                                  % error_msg, logging.info)
+            if not re.search(error_msg, output):
+                test.fail("Can not get expected error message: %s from %s"
+                          % (error_msg, output))
+        except qemu_monitor.MonitorConnectError as detail:
```
Comment (PaulYuuu) [inline qemu/tests/power_nvram.py:34]:
```suggestion
        except qemu_monitor.MonitorConnectError:
```

BTW, why pass `MonitorConnectError` directly?

Comment (MiriamDeng) [reply qemu/tests/power_nvram.py]: Updated it and the second test case took much time to pop the warnings but it already started connecting monitor. Do you have any better solution for this situation? Thanks.

**Comment by MiriamDeng** [comment]
Update the patch, could please help to review it again ? Thanks a lot. @PaulYuuu @maxujun

**Comment by maxujun** [comment]
# git am 2687.patch
Applying: power_nvram: make sure qemu-kvm works well while raw strings are passed
.git/rebase-apply/patch:16: trailing whitespace.
    start_vm = no
.git/rebase-apply/patch:18: trailing whitespace.
    # the below test cases make sure qemu won't cordump
warning: 2 lines add whitespace errors.

**Comment by MiriamDeng** [comment]
Update, @maxujun could you please help to check it again ? Thanks a lot.

File: qemu/tests/power_nvram.py (hunk 2 above)
Comment (PaulYuuu) [reply qemu/tests/power_nvram.py]: Could you try to increase the timeout when creating the VM?
```
vm.create(params=params, timeout=timeout)
```

Comment (MiriamDeng) [reply qemu/tests/power_nvram.py]: It doesn't work even if the timeout was set to.

Comment (PaulYuuu) [reply qemu/tests/power_nvram.py]: Okay, you can use `debugonly = yes`.
Besides, please add `only ppc64 ppc64le`

Comment (MiriamDeng) [reply qemu/tests/power_nvram.py]: Updated it and keep the original way as we discussed yesterday, thanks.

**Review by PaulYuuu — APPROVED** [review]

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**